### PR TITLE
Updated CHANGELOG.md for 0.20.1 release.

### PR DESCRIPTION
**Pull request by mingjie-intel**
- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,20 @@ All notable changes to this project will be documented in this file.
...
-## [Unreleased] - 2023-03-28
+## [0.20.1] - 2023-04-07
 
 ### Added
-* Replaced llvm_spirv from oneAPI path by dpcpp-llvm-spirv package.
+* Replaced llvm_spirv from oneAPI path by dpcpp-llvm-spirv package.(#979)
+* Added Dockerfile and a manual workflow to publish pre-built packages to the repo.(#973)
+
+### Changed
+* Allowed shape to be a positional only argument when constructing a tensor. (#993)
```
Comment (diptorupd) [inline CHANGELOG.md:14]: #993 should go under fixed. (Fixed default dtype derivation when creating a dpnp.ndarray)

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,10 +4,20 @@ All notable changes to this project will be documented in this file.
...
+* Adjusted test_windows step to work with intel-opencl-rt=2023.1.0. (#990)
+* Handled the case when arraystruct->meminfo is null to close gh-965. (#972)
```
Comment (diptorupd) [inline CHANGELOG.md:16]: Should go under fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,10 +4,20 @@ All notable changes to this project will be documented in this file.
...
+* Refactored kernel_builder.py. (#967)
```
Comment (diptorupd) [inline CHANGELOG.md:17]: Is this one worth mentioning?

Comment (mingjie-intel) [reply CHANGELOG.md]: > Is this one worth mentioning?

Removed.

File: CHANGELOG.md (hunk 2 above)
Comment (mingjie-intel) [reply CHANGELOG.md]: > Should go under fixed.

Moved.

File: CHANGELOG.md (hunk 1 above)
Comment (mingjie-intel) [reply CHANGELOG.md]: > Reference in new

Moved.

**Review by diptorupd — APPROVED** [review]

**Pull request merged**